pynest/examples/Potjans_2014/helpers.py

```diff
@@ -19,315 +19,176 @@
 
 
-def compute_DC(net_dict, w_ext):
-    """ Computes DC input if no Poisson input is provided to the microcircuit.
+def num_synapses_from_conn_probs(conn_probs, popsize1, popsize2):
+    """Computes the total number of synapses between two populations from
+    connection probabilities.
 
-    Parameters
-    ----------
-    net_dict
-        Parameters of the microcircuit.
-    w_ext
-        Weight of external connections.
+    Here it is irrelevant which population is source and which target.
+
+    Paramters
+    ---------
+    conn_probs
+        Matrix of connection probabilities.
+    popsize1
+        Size of first poulation.
+    popsize2
+        Size of second population.
 
     Returns
     -------
-    DC
-        DC input, which compensates lacking Poisson input.
+    num_synapses
+        Matrix of synapse numbers.
     """
-    DC = (
-        net_dict['bg_rate'] * net_dict['K_ext'] *
-        w_ext * net_dict['neuron_params']['tau_syn_E'] * 0.001
-        )
-    return DC
+    prod = np.outer(popsize1, popsize2)
+    num_synapses = np.log(1. - conn_probs) / np.log((prod - 1.) / prod)
+    return num_synapses
 
 
-def get_weight(PSP_val, net_dict):
-    """ Computes weight to elicit a change in the membrane potential.
+def weight_as_current_from_potential(PSP, C_m, tau_m, tau_syn):
+    """ Computes weight as postsynaptic current from postsynaptic potential.
 
-    This function computes the weight which elicits a change in the membrane
-    potential of size PSP_val. To implement this, the weight is calculated to
-    elicit a current that is high enough to implement the desired change in the
-    membrane potential.
+    The weight is calculated as a postsynaptic current that is large enough to
+    elicit a change in the membrane potential of size ``PSP_e``.
 
     Parameters
     ----------
-    PSP_val
-        Evoked postsynaptic potential.
-    net_dict
-        Dictionary containing parameters of the microcircuit.
+    PSP
+        Mean amplitude of postsynaptic potential (in mV).
+    C_m
+        Membrane capacitance (in pF).
+    tau_m
+        Membrane time constant (in ms).
+    tau_syn
+        Synaptic time constant (in ms).
 
     Returns
     -------
-    PSC_e
-        Weight value(s).
+    PSC
+        Amplitude of postynaptic current (in pA).
 
     """
-    C_m = net_dict['neuron_params']['C_m']
-    tau_m = net_dict['neuron_params']['tau_m']
-    tau_syn_ex = net_dict['neuron_params']['tau_syn_ex']
-
-    PSC_e_over_PSP_e = (((C_m) ** (-1) * tau_m * tau_syn_ex / (
-        tau_syn_ex - tau_m) * ((tau_m / tau_syn_ex) ** (
-            - tau_m / (tau_m - tau_syn_ex)) - (tau_m / tau_syn_ex) ** (
-                - tau_syn_ex / (tau_m - tau_syn_ex)))) ** (-1))
-    PSC_e = (PSC_e_over_PSP_e * PSP_val)
-    return PSC_e
-
-
-def get_total_number_of_synapses(net_dict):
-    """ Returns the total number of synapses between all populations.
+    PSC_over_PSP = (((C_m) ** (-1) * tau_m * tau_syn / (
```

Comment:
As far as I know, this equation is not mentioned in any paper, yet. I revised the code to make it more readable and added an explanation on how the equation is obtained in the docstring.

Reply:
The expression for tmax appears in the appendix of Hanuschkin et al. (2010). From this V(tmax) can be easily computed. The Hanuschkin et al. (2010) reference is the best we can do for now. Also Krishnan et al. (2018) cites Hanuschkin in this way.

Hanuschkin A, Kunkel S, Helias M, Morrison A and Diesmann M (2010) A general and efficient method for incorporating precise spike times in globally time-driven simulations. Front. Neuroinform. 4:113. doi: 10.3389/fninf.2010.00113

Krishnan J, Porta Mana P, Helias M, Diesmann M and Di Napoli E (2018) Perfect Detection of Spikes in the Linear Sub-threshold Dynamics of Point Neurons. Front. Neuroinform. 11:75. doi: 10.3389/fninf.2017.00075

Reply:
Thanks, @diesmann! I added the reference. In addition, I changed the function so that it returns the conversion factor `PSC_over_PSP` and has to be called only once. Please check.